src/osd_window.rs

```diff
@@ -175,18 +175,41 @@ impl SwayosdWindow {
 		self.run_timeout();
 	}
 
-	pub fn changed_capslock(&self, state: bool) {
+	pub fn changed_keylock(&self, key: KeysLocks, state: bool) {
 		self.clear_osd();
 
-		let icon = self.build_icon_widget("input-keyboard-symbolic");
 		let label = self.build_text_widget(None);
 
+		let on_off_text = match state {
+			true => "On",
+			false => "Off",
+		};
+
+		let (label_text, symbol) = match key {
+			KeysLocks::CapsLock => {
+				let symbol = "input-keyboard-symbolic";
+				let text = "Caps Lock ".to_string() + on_off_text;
+				(text, symbol)
+			}
+			KeysLocks::NumLock => {
+				let symbol = "input-keyboard-numlock-symbolic";
```

Comment:
Seems like these new icons aren't available in the default Adwaita icon theme. We should include some icons with swayosd. That would require a meson build file + other things. Would you mind if I pushed to your branch adding these?

Reply:
I've updated the main branch with a way of using local icons. Just let me know if you don't mind me pushing a few icons :)

Reply:
I have no issues with you pushing icons :)
Sorry for the late answer, I cut a small part of my finger of so I wasn't at my PC.

Reply:
> Sorry for the late answer, I cut a small part of my finger off so I wasn't at my PC.

Sorry to hear that. Get well! :)